proxy.go

```diff
@@ -16,15 +18,28 @@ type proxy struct {
 	proxy       *httputil.ReverseProxy
 }
 
-func newProxy(backendURL, emailHeader string) (*proxy, error) {
+func newProxy(backendURL, emailHeader string, timeout time.Duration) (*proxy, error) {
 	backend, err := url.Parse(backendURL)
 	if err != nil {
 		return nil, fmt.Errorf("Could not parse URL '%s': %s", backendURL, err)
 	}
+	p := httputil.NewSingleHostReverseProxy(backend)
```

Comment:
Overall LGTM but I think it would be better if we could have the default transport settings and only override the `Timeout` setting. Do you think that's easy to do?